internal/resolver/delegatingresolver/delegatingresolver.go

```diff
@@ -196,11 +197,38 @@ func (r *delegatingResolver) Close() {
 // either resolver has not sent update even once and returns the error from
 // ClientConn update once both resolvers have sent update atleast once.
 func (r *delegatingResolver) updateClientConnStateLocked() error {
-	if r.targetResolverState == nil || r.proxyAddrs == nil {
+	if r.targetResolverState == nil {
 		return nil
 	}
 
 	curState := *r.targetResolverState
+	isTCP := false
+
+	for _, addr := range curState.Addresses {
+		if networkType, ok := networktype.Get(addr); !ok || networkType == "tcp" {
+			isTCP = true
+			break
+		}
```

Comment:
Out here we're assuming `!ok` to be the same of `tcp`. However, the old code in the dialer used to parse the address string when `!ok` to determine the network type, we should probably maintain the same behaviour.
https://github.com/grpc/grpc-go/blob/172fc5bbb4618bc910e875823cf964f03f1305b1/internal/transport/http2_client.go#L177-L182